File: python/src/main/python/store/YogurtIceCreamStore.py
from icecream.yogurt.GuavaYogurtIceCream import *
from icecream.yogurt.MangoYogurtIceCream import *
from icecream.yogurt.PineappleYogurtIceCream import *
from icecream.yogurt.StrawberryYogurtIceCream import *
from IceCreamStore import *


class YogurtIceCreamStore(IceCreamStore):
    def createIceCream(self, selectedFlavor):
        if (selectedFlavor == "Strawberry"):
            return StrawberryYogurtIceCream()
        elif (selectedFlavor == "Pineapple"):
            return PineappleYogurtIceCream()
        elif (selectedFlavor == "Mango"):
            return MangoYogurtIceCream()
        elif (selectedFlavor == "Guava"):
            return GuavaYogurtIceCream()
        return None

File: java/src/main/java/mx/iteso/factory/stores/YogurtIceCreamStore.java
package mx.iteso.factory.stores;

import mx.iteso.factory.IceCreamStore;
import mx.iteso.factory.icecreams.IceCream;
import mx.iteso.factory.icecreams.yogurt.GuavaYogurtIceCream;
import mx.iteso.factory.icecreams.yogurt.MangoYogurtIceCream;
import mx.iteso.factory.icecreams.yogurt.PineappleYogurtIceCream;
import mx.iteso.factory.icecreams.yogurt.StrawberryYogurtIceCream;

/** Yogurt Ice Cream store class.*/
public class YogurtIceCreamStore extends IceCreamStore {

    /**
     * Create ice cream.
     * @param selectedFlavor the flavor.
     * @return IceCream.
     */
    @Override
    protected IceCream createIceCream(final String selectedFlavor) {
        if (selectedFlavor.equals("Strawberry")) {
            return new StrawberryYogurtIceCream();
        } else if (selectedFlavor.equals("Pineapple")) {
            return new PineappleYogurtIceCream();
        } else if (selectedFlavor.equals("Mango")) {
            return new MangoYogurtIceCream();
        } else if (selectedFlavor.equals("Guava")) {
            return new GuavaYogurtIceCream();
        }
        return null;
    }
}
